notebook/static/notebook/js/menubar.js

```diff
@@ -189,6 +189,27 @@ define([
             that._nbconvert('html', false);
         });
 
+        this.element.find('#document_details').click(function () {
+            var months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
+            var table = $('<table></table>');
+            var body = $('<tbody></tbody>');
+            
+            var lmod = that.notebook.last_modified;
+            var last_modified_str = months[lmod.getMonth()] + ' ' +
+                lmod.getDate() + ', ' + lmod.getFullYear() + ' ' + lmod.getHours() + ':'+ lmod.getMinutes();
```

Comment:
We use a package called `moment.js` to format dates - there are different conventions in different places. Have a look at the code for the notebook list here: https://github.com/jupyter/notebook/blob/d9baafaabf23d0c0ba7a05254083f69efc398cb9/notebook/static/tree/js/notebooklist.js#L865-L867